libraries/math/src/hierarchical-clustering/wasm/cpp/fastcluster.cpp

```diff
@@ -0,0 +1,191 @@
+//
```

Comment:
I think that we should discuss with @skalkin the following:
are we allowed to present such files in our repository?

Reply:
we had a discussion actually when I was adding this in dendrogram package. we decided to preserve the code along with its license. as long as its not too large